packages/components/CONTRIBUTING.md

```diff
@@ -4,12 +4,202 @@ Thank you for taking the time to contribute.
+
+#### Emotion
+
+All new component should be styled using [Emotion](https://emotion.sh/docs/introduction).
+
+Note: Instead of using Emotion's standard `cx` function, the custom [`useCx` hook](/packages/components/src/utils/hooks/use-cx.ts) should be used instead.
+
+#### Context system
+
+The `@wordpress/components` context system is based on [React's `Context` API](https://reactjs.org/docs/context.html), and is a way for components to adapt to the "context" they're being rendered in.
+
+Components can use this system via a couple of functions:
+
+- they can provide values using a shared `ContextSystemProvider` component
+- they can connect to the Context via `contextConnect`
+- they can read the "computed" values from the context via `useContextSystem`
+
+An example of how this is used is the [`Card` component](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/card/card/component.js#L44-L54) — all of its subcomponents (e.g. `CardBody`, `CardHeader`...) will [use, by default, the same value for the `size` as the parent `Card` component](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/card/card-body/hook.js#L23) — which results in all the components having all the correct spacing "auto-magically".
+
+#### Unit tests
+
+Please refer to the [JavaScript Testing Overview docs](/docs/contributors/code/testing-overview.md).
+
+#### Storybook
+
+All new components should include [Storybook](https://storybook.js.org/) stories, in order to make it easier to work on and review each component in isolation.
+
+A component's stories should be showcasing its different states — for example, a `Button`'s different variants:
+
+```jsx
+import Button from '../';
+
+export default { title: 'Components/Button', component: Button };
+
+export const _default = () => <Button>Default Button</Button>;
+
+export const primary = () => <Button variant="primary">Primary Button</Button>;
+
+export const secondary = () => <Button variant="secondary">Secondary Button</Button>;
+```
+
+For even more flexibility while writing stories, the [`@storybook/addons-controls`](https://storybook.js.org/addons/@storybook/addon-controls) addon is a great way to interact with a component's arguments dynamically and in real time.
+
+Ideally, all of the props for the story's components should be exposed by using knobs. The default value of the knobs should coincide with the default value of the props (i.e. it should be `undefined` if a prop is not required). A story should, therefore, also explicitely show how values from the Context System are applied to (sub)components. A good example of how this may look like is the [`Card` story](https://wordpress.github.io/gutenberg/?path=/story/components-card--default) (code [here](/packages/components/src/card/stories/index.js)).
```

Comment:
```suggestion
Ideally, all of the props for the story's components should be exposed by using knobs. The default value of the knobs should coincide with the default value of the props (i.e. it should be `undefined` if a prop is not required). A story should, therefore, also explicitly show how values from the Context System are applied to (sub)components. A good example of how this may look like is the [`Card` story](https://wordpress.github.io/gutenberg/?path=/story/components-card--default) (code [here](/packages/components/src/card/stories/index.js)).
```